src/utils/action.js

```diff
@@ -56,17 +56,19 @@ function getInput(name, required = false) {
 /**
  * Executes the provided shell command
  * @param {string} cmd - Shell command to execute
- * @param {{dir: string, ignoreErrors: boolean}} [options] - {@see RUN_OPTIONS_DEFAULTS}
+ * @param {{dir: string, ignoreErrors: boolean, prefix: string}} [options] - {@see RUN_OPTIONS_DEFAULTS}
  * @returns {{status: number, stdout: string, stderr: string}} - Output of the shell command
  */
 function run(cmd, options) {
+	const prefix = options ? options.prefix : "";
+	const command = `${prefix}${cmd}`;
```

Comment:
Do we need the `prefix` option? The `run` function should be as general as possible and could be called with the concatenated string.

Reply:
I was trying to avoid having every single linter implement what ends up being exactly the same behavior and centralizing it. There isn't anything linter-specific in this functionality. I can extract this part into another function, but I'm not sure how much we'd be gaining.

Reply:
I'd prefer to have `commandPrefix` as another parameter in the `lint` function and then have it added to the lint command like `${commandPrefix} eslint ...`. This keeps the implementation consistent with the other linter-specific options.

Reply:
@samuelmeuli started looking at doing it this way and then realized that this would mean that the base `action` file would need to know whether to use the `prefix` or `npmPrefix` function depending on the language, which doesn't seem very clean.

Maybe a better approach would be to define a `commandPrefix` function on the class, and use that inside the `lint` and `verifySetup` functions? We could either have the `action` file call the `commandPrefix` function and pass it in, or have it used internally within those 2 functions.

Reply:
What I'm thinking is that `index.js` would simply pass the option value to the linter. Most linters can use that value as-is, Node.js linters could call `getNpmPrefix` if the value is `""`. This may not be the most concise way of implementing this, but I think it's more readable and consistent with how the other options are implemented. If there's a nicer way, I think it should be applied to all options in a separate change in the future.

Regarding the second approach you're mentioning: Isn't the `lint` function the only place where the prefix needs to be used? Keep in mind that the option value is not necessarily something like `bundler exec`, but could also be used like described [here](https://github.com/samuelmeuli/lint-action/issues/20#issuecomment-590498256).

Reply:
I think it will need to be passed into `verifySetup` as well. JS linters already use `npmPrefix` in there and we've defined the `command_prefix` option as overriding that value.